src/tests/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess.AcceptanceTests/Services/TestPaymentsProcessFunctions.cs

```diff
@@ -43,163 +47,107 @@ public void Dispose()
         protected virtual void Dispose(bool disposing)
         {
             if (_isDisposed) return;
-
-            if (_functionHostProcess != null)
-            {
-                if (!_functionHostProcess.HasExited)
-                {
-                    _functionHostProcess.Kill();
-                }
-
-                _functionHostProcess.CloseMainWindow();
-                _functionHostProcess.Dispose();
-            }
-
-            _testContext.SqlDatabase?.Dispose();
+            _functionsController.Dispose();
+            _client.Dispose();
             _isDisposed = true;
         }
 
-        public async Task<AzureFunctionOrchestrationStatus> StartPaymentsProcess(short collectionPeriodYear, byte collectionPeriodMonth)
+        public async Task<OrchestrationStatus> StartPaymentsProcess(short collectionPeriodYear, byte collectionPeriodMonth)
         {
-            var orchestrationLinks = await StartFunctionOrchestration(collectionPeriodYear, collectionPeriodMonth);
-            return await CompleteFunctionOrchestration(orchestrationLinks);
+            var orchestrationLinks = await StartIncentivePaymentOrchestrator(collectionPeriodYear, collectionPeriodMonth);
+            return await FunctionOrchestrationCompleted(orchestrationLinks);
         }
 
         public async Task StartLearnerMatching()
         {
             await StartLearnerMatchingOrchestrator();
-            Thread.Sleep(TimeSpan.FromSeconds(10));
+            Thread.Sleep(TimeSpan.FromSeconds(1)); // time it takes function host to update storage 🤷‍
```

Comment:
Can we ping the health check instead to it's not a random thread.sleep?

Reply:
we *are* using status API (see AllFunctionOrchestrationCompleted) however, it takes a bit of time for a function run to show up in there apparently (in any status) . so this is a necessity unfortunately.